p2p/src/authenticated/discovery/actors/peer/actor.rs

```diff
@@ -261,9 +272,11 @@ impl<E: Spawner + Clock + ReasonablyRealtime + Rng + CryptoRng + Metrics, C: Pub
             });
 
         // Wait for one of the handlers to finish
-        //
-        // It is only possible for a handler to exit if there is an error.
+        let mut shutdown = self.context.stopped();
         let result = select! {
+            _ = &mut shutdown => {
```

Comment:
I think we could just have shutdown here (rather than in 3 places)?

Reply:
Done. To make this work, I did have to make [this change](https://github.com/commonwarexyz/monorepo/pull/2333/commits/2b6f328dd50ada576da20c09e797333925a7b616)